### PR TITLE
feat: modernize codebase using Java switch expressions and List#getFirst/getLast APIs

### PR DESCRIPTION
**Pull request by Ludy87**
# Description of Changes

This PR introduces a broad modernization of the codebase by adopting newer Java language features and improving code readability and maintainability.

## What was changed

- Replaced traditional `switch` statements with modern switch expressions (`case ->`) across multiple classes.
- Replaced usages of `List#get(0)` and `List#get(size - 1)` with `getFirst()` and `getLast()` respectively.
- Simplified conditional logic using pattern matching (e.g., `instanceof` and switch pattern matching).
- Refactored various utility and controller classes to reduce boilerplate and improve clarity.
- Removed unused or redundant code (e.g., `parseClientFileIds` method in `MergeController`).
- Improved type safety (e.g., using `Class::isInstance` instead of `instanceof` checks in streams).
- Cleaned up Spring annotations by removing unnecessary `@Autowired` where constructor injection is already used.
- Added a new test (`UIDataControllerTest`) to ensure correct handling of identical JSON configs with different filenames.
- Minor formatting and style fixes (e.g., Spotless formatting adjustment).

## Why the change was made

- To align the codebase with modern Java standards (Java 17+ features).
- To improve readability and maintainability by reducing verbosity.
- To eliminate common indexing patterns that are more error-prone.
- To standardize coding style across the project.
- To improve test coverage for edge cases discovered during refactoring.

---

### General

- [ ] I have read the [Contribution Guidelines](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/CONTRIBUTING.md)
- [ ] I have read the [Stirling-PDF Developer Guide](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/DeveloperGuide.md) (if applicable)
- [ ] I have read the [How to add new languages to Stirling-PDF](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/devGuide/HowToAddNewLanguage.md) (if applicable)
- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings

### Documentation

- [ ] I have updated relevant docs on [Stirling-PDF's doc repo](https://github.com/Stirling-Tools/Stirling-Tools.github.io/blob/main/docs/) (if functionality has heavily changed)
- [ ] I have read the section [Add New Translation Tags](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/devGuide/HowToAddNewLanguage.md#add-new-translation-tags) (for new translation tags only)

### Translations (if applicable)

- [ ] I ran [`scripts/counter_translation.py`](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/docs/counter_translation.md)

### UI Changes (if applicable)

- [ ] Screenshots or videos demonstrating the UI changes are attached (e.g., as comments or direct attachments in the PR)

### Testing (if applicable)

- [ ] I have run `task check` to verify linters, typechecks, and tests pass
- [ ] I have tested my changes locally. Refer to the [Testing Guide](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/DeveloperGuide.md#7-testing) for more details.

### CODE REVIEW

File: CLAUDE.md~Stashed changes [hunk 1]
```diff
@@ -0,0 +1 @@
+AGENTS.md
```
Comment (Frooodle) [inline CLAUDE.md~Stashed changes]: CLAUDE.md~Stashed changes ?

**Review by Frooodle — CHANGES_REQUESTED** [review]

File: CLAUDE.md~Stashed changes (hunk 1 above)
Comment (Ludy87) [reply CLAUDE.md~Stashed changes]: > CLAUDE.md~Stashed changes ?

sorry

File: app/saas/src/main/java/stirling/software/saas/payg/api/PaygInvoicesController.java [hunk 2]
```diff
@@ -105,7 +105,7 @@ public ResponseEntity<List<InvoiceResponse>> list(
         if (rows.isEmpty()) {
             return ResponseEntity.ok(List.of());
         }
-        Long teamId = rows.get(0).getTeam().getId();
+        Long teamId = rows.getFirst().getTeam().getId();
```
[the remark this answers is not in the document]
Comment (Ludy87) [reply app/saas/src/main/java/stirling/software/saas/payg/api/PaygInvoicesController.java:108]: @AikidoSec feedback:  List#getFirst() is valid in this repository because the project targets Java 25, and List implements SequencedCollection there. The repo already uses getFirst() in production code, so this change is consistent. The test update is intentional and should compile under the configured toolchain.

File: app/saas/src/main/java/stirling/software/saas/accountlink/AccountLinkController.java [hunk 3]
```diff
@@ -152,7 +152,7 @@ private LeaderTeam resolveLeaderTeam(Authentication auth) {
         if (rows.isEmpty()) {
             return new LeaderTeam(null, null, HttpStatus.FORBIDDEN);
         }
-        TeamMembership m = rows.get(0);
+        TeamMembership m = rows.getFirst();
```
[the remark this answers is not in the document]
Comment (Ludy87) [reply app/saas/src/main/java/stirling/software/saas/accountlink/AccountLinkController.java:155]: @AikidoSec feedback:  List#getFirst() is valid in this repository because the project targets Java 25, and List implements SequencedCollection there. The repo already uses getFirst() in production code, so this change is consistent. The test update is intentional and should compile under the configured toolchain.

File: app/saas/src/main/java/stirling/software/saas/payg/api/PaygWalletController.java [hunk 4]
```diff
@@ -334,7 +334,7 @@ public record UpdateCapRequest(@Min(0) int capUsd, boolean noCap) {}
 
     private Optional<TeamMembership> primaryMembership(Long userId) {
         List<TeamMembership> rows = memberRepo.findPrimaryMembership(userId);
-        return rows.isEmpty() ? Optional.empty() : Optional.of(rows.get(0));
+        return rows.isEmpty() ? Optional.empty() : Optional.of(rows.getFirst());
```
[the remark this answers is not in the document]
Comment (Ludy87) [reply app/saas/src/main/java/stirling/software/saas/payg/api/PaygWalletController.java:430]: @AikidoSec feedback:  This repo targets Java 25, and List#getFirst() is available there via SequencedCollection. The change is intentional and consistent with the repository’s existing Java 25 setup, so rows.getFirst() should compile here.

File: app/saas/src/main/java/stirling/software/saas/payg/api/PaygPaymentMethodController.java [hunk 5]
```diff
@@ -85,7 +85,7 @@ public ResponseEntity<PaymentMethodResponse> get(Authentication auth) {
         if (rows.isEmpty()) {
             return ResponseEntity.ok(PaymentMethodResponse.absent());
         }
-        Long teamId = rows.get(0).getTeam().getId();
+        Long teamId = rows.getFirst().getTeam().getId();
```
[the remark this answers is not in the document]
Comment (Ludy87) [reply app/saas/src/main/java/stirling/software/saas/payg/api/PaygPaymentMethodController.java:88]: @AikidoSec feedback:  This repo targets Java 25, and List#getFirst() is available there via SequencedCollection. The change is intentional and consistent with the repository’s existing Java 25 setup, so rows.getFirst() should compile here.

**Review by Frooodle — DISMISSED** [review]

**Review by ConnorYoh — DISMISSED** [review]

**Review by Frooodle — APPROVED** [review]

**Review by ConnorYoh — APPROVED** [review]

**Review by jbrunton96 — APPROVED** [review]
